iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs

```diff
@@ -36,6 +36,18 @@ internal class AmqpConnectionPool : IAmqpUnitManager
                 {
                     AmqpConnectionHolder[] amqpConnectionHolders = ResolveConnectionGroup(deviceIdentity);
                     amqpConnectionHolder = ResolveConnectionByHashing(amqpConnectionHolders, deviceIdentity);
+
+                    if (deviceIdentity.AuthenticationModel == AuthenticationModel.SasGrouped)
+                    {
+                        // For group sas token authenticated devices over a multiplexed connection, the identity
+                        // of the first device connecting will be used for generating the group sas tokens.
+                        // The TokenRefresher of the subsequently connected device identities can be safely disposed.
+                        if (!ReferenceEquals(amqpConnectionHolder.GetDeviceIdentityOfAuthenticationProvider(), deviceIdentity)
```

Comment:
Can you connect this logic to the comments above? I don't see the connection.

Reply:
And why doesn't it include the `ShouldSdkDisposeInstance` check?

Reply:
If I go with this flow, I'll add some additional comments corelating the logic.